python/setup.py

```diff
@@ -42,24 +43,33 @@ class Package(NamedTuple):
     lib_flag: str
     syspath_var_name: str
 
+# pybind11
+
 
 def get_pybind11_package_info():
     name = "pybind11-2.10.0"
     url = "https://github.com/pybind/pybind11/archive/refs/tags/v2.10.0.tar.gz"
     return Package("pybind11", name, url, "include/pybind11/pybind11.h", "PYBIND11_INCLUDE_DIR", "", "PYBIND11_SYSPATH")
 
+# llvm
+
 
 def get_llvm_package_info():
     # download if nothing is installed
     system = platform.system()
-    system_suffix = {"Linux": "linux-gnu-ubuntu-18.04", "Darwin": "apple-darwin"}[system]
-    use_assert_enabled_llvm = check_env_flag("TRITON_USE_ASSERT_ENABLED_LLVM", "False")
-    if use_assert_enabled_llvm:
-        name = 'llvm+mlir-14.0.0-x86_64-{}-assert'.format(system_suffix)
-        url = "https://github.com/shintaro-iwasaki/llvm-releases/releases/download/llvm-14.0.0-329fda39c507/{}.tar.xz".format(name)
+    if system == "Darwin":
+        system_suffix = "apple-darwin"
+    elif system == "Linux":
+        vglibc = tuple(map(int, platform.libc_ver()[1].split('.')))
+        vglibc = vglibc[0] * 100 + vglibc[1]
+        linux_suffix = 'ubuntu-18.04' if vglibc > 217 else 'centos-7'
+        system_suffix = f"linux-gnu-{linux_suffix}"
     else:
-        name = 'clang+llvm-14.0.0-x86_64-{}'.format(system_suffix)
-        url = "https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/{}.tar.xz".format(name)
+        raise RuntimeError(f"unsupported system: {system}")
+    use_assert_enabled_llvm = check_env_flag("TRITON_USE_ASSERT_ENABLED_LLVM", "False")
+    release_suffix = "assert" if use_assert_enabled_llvm else "release"
+    name = f'llvm+mlir-14.0.6-x86_64-{system_suffix}-{release_suffix}'
+    url = f"https://github.com/ptillet/triton-llvm-releases/releases/download/llvm-14.0.6-f28c006a5895/{name}.tar.xz"
```

Comment:
Notice that this PR pulls a modified LLVM 14.0 build.

Reply:
Looks like triton maintainer built and supplied llvm 14 built on centos7.  I'm guessing pytorch maintainers asked for it.

Reply:
Probably yes.  Triton is getting ready to do a prerelease of Triton 2.0 this week.  This PR is probably necessary to make that happen.